utils/warnings.ml

```diff
@@ -91,6 +91,7 @@ type t =
   | Unsafe_without_parsing                  (* 64 *)
   | Redefining_unit of string               (* 65 *)
   | Unused_open_bang of string              (* 66 *)
+  | Generic_comparison                      (* 67 *)
```

Comment:
I think it would be nicer to include the offending operation in question, perhaps like this:
```ocaml
| Generic_comparison of string
```

and then give more precise warnings:

```
this comparison will use the generic equality function
this comparison will use the generic comparison function
this comparison will use the generic inequality function
```
rather than calling everything "comparison".  (For one thing, it's not quite accurate to say "the" generic comparison function, but I think it's generally better to be as precise as possible, if it's not a lot of extra work.)